server/lib/init_log.rb

```diff
@@ -0,0 +1,24 @@
+require "fusor/multilog"
+
+module Fusor
+  @default_log_file = "/var/log/foreman/fusor.log"
+
+  def self.log
+    @log ||= MultiLogger.new(Rails.logger)
+  end
+
+  def self.log_change_deployment(deployment = nil)
+    self.log
+
+    #@log.detach_all
+
+    if Rails.env.production?
+      if deployment.nil?
+        @log.attach(@default_log_file)
+      else
+        Dir.mkdir("/var/log/foreman/#{deployment.name}-#{deployment.id}") unless File.exist?("/var/log/foreman/#{deployment.name}-#{deployment.id}")
+        @log.attach("/var/log/foreman/#{deployment.name}-#{deployment.id}/deployment.log")
```

Comment:
for the deployment.name issues seems like we can do 1 of 2 things pretty easily.
1) just drop name and use id which should always be unique
or 2) strip out any potentially offensive chars.
```ruby
deployment.name.gsub!(/[^0-9A-Za-z.\-]/, '_')
```